metagraph/src/graph/representation/succinct/boss_chunk.cpp

```diff
@@ -138,7 +138,8 @@ BOSS::Chunk::Chunk(uint64_t alph_size, size_t k, bool canonical,
 }
 
 BOSS::Chunk::~Chunk() {
-    utils::remove_temp_dir(dir_);
+    if (!dir_.empty())
+        utils::remove_temp_dir(dir_);
```

Comment:
So I'm guessing the logic here is that if the tmp directory is corrupted somehow you want to keep it? Maybe clarify this with a comment.

Reply:
Not exactly. Now we can have uninitialized chunks, where dir_ is empty, and thus can't call blindly call `utils::remove_temp_dir` which can only be called for directories created by `utils::create_temp_dir`.

Reply:
I added a comment